app/cdap/components/GridTable/components/NestedMenu/index.tsx

```diff
@@ -31,34 +31,36 @@ export default function({
   setAnchorEl,
   open,
   handleMenuOpenClose,
-}: INestedMenuProps) {
-  const [anchorEl2, setAnchorEl2] = useState<EventTarget | null>(null);
-  const [nestedOptions, setNestedOptions] = useState<IMenuItem[]>([]);
+  newIndex,
+}: INestedMenuProps) => {
+  const [menuComponentOptions, setMenuComponentOptions] = useState([]);
   const classes = useNestedMenuStyles();
 
   const handleMenuClick = (
     event: React.MouseEvent<HTMLLIElement, MouseEvent>,
     menuItem: IMenuItem
   ) => {
-    setNestedOptions([]);
     event.preventDefault();
     event.stopPropagation();
-    if (menuItem?.options) {
-      setNestedOptions(menuItem.options);
-      setAnchorEl2(event.currentTarget);
+    if (menuItem?.options?.length > 0) {
```

Comment:
this component should render any number of nested menus as long as we send some options

Reply:
No need of using hasOptions, we are checking it with the help of options length